File: MyCS50 code/DanilaDz-cs50-problems-2020-x-readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main(void)
{
    // input
    string input = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;
    // counts words
    for (int spaces = 0; spaces < strlen(input); spaces++)
    {
        if (input[spaces] == ' ')
        {
            words++;
        }
    }
// counts letters
    for (int letter = 0; letter < strlen(input); letter++)
    {
        if (tolower(input[letter]) >= 'a' && tolower(input[letter]) <= 'z')
        {
            letters++;  
        }
            
    }
    // counts sentences
    for (int sent = 0; sent < strlen(input); sent++)
    {
        if (input[sent] == '.' || input[sent] == '!' || input[sent] == '?')
        {
            sentences++;
        }
    }
    // avarages the amount of letters and sentences per 100 words
    float L = (letters / ((float) words)) * 100;
    float S = (sentences / ((float) words)) * 100;
    int liau_index = round(0.0588 * L - 0.296 * S - 15.8);
    
    // prints out the answer
    if (liau_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (liau_index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", liau_index);
    }
    
}

File: MyCS50 code/DanilaDz-cs50-problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string
import math


def main():
    
    # gets text from user
    text = get_string('Text: ').upper()
    letters = 0
    words = 1
    sentences = 0
    
    # counts letters
    for letter in text:
        if letter >= 'A' and letter <= 'Z':
            letters += 1
    
    # counts words
    for word in text:
        if word == ' ':
            words += 1
    
    # counts sentences
    for sentence in text:
        if sentence == '.' or sentence == '!' or sentence == '?':
            sentences += 1
    
    # does liau formula
    L = float(letters / words * 100)
    S = float(sentences / words * 100)
    liau = int(round(0.0588 * L - 0.296 * S - 15.8))
    
    # checks for grade level
    if liau < 1:
        print("Before Grade 1")
    
    elif liau >= 16:
        print("Grade 16+")
    
    else:
        print("Grade ", liau)


main()
